src/hooks/useContract.ts

```diff
@@ -16,4 +18,22 @@ export default function useContract(address, abi) {
   }, [library, abi, address]);
 
   return resolver;
-}
+};
+
+export const useContractCall = (
+  address: string,
+  abi: any[],
+  functionName: string,
+  params: string[]
+): string => {
+  const { active, library } = useWeb3React();
+  const { data } = useSWR(id(address + functionName + params), async () => {
+    if (!active) return '0x0';
+
+    const provider = new providers.Web3Provider(library.currentProvider);
+    const tokenContract = getContract(address, abi, provider);
+    return await tokenContract[functionName](...params);

```

Comment:
I don't understand what you are trying to do here exactly?

Reply:
A hook to do contract call, with the address, abi, function name and funtion params.

The value fetched is saved in the SWR cache with a deterministic id, the value fetched is saved on the SWR cache on the id with the hash of the contract address, function and params, which means that the only way to get the stored value is doing the same call to the same smart contract again.

Reply:
Later it would be good to see how we can use typechain contracts.

Reply:
ok, I understand now. `id(..)` is just hashing the provided parameters. we can do the refactoring of this at a later date if you wish so.